keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/SumVertexTest.java

```diff
@@ -1,32 +1,125 @@
 package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.Differentiator;
 import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.PartialDerivatives;
 import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
-import org.junit.Test;
-
-import static org.junit.Assert.assertArrayEquals;
-import static org.junit.Assert.assertEquals;
 
 public class SumVertexTest {
 
     @Test
     public void doesSum() {
+        DoubleVertex a = new UniformVertex(new int[]{1, 5}, 0, 10);
+        a.setValue(new double[]{1, 2, 3, 4, 5});
+
+        DoubleVertex summed = a.sum();
+
+        assertEquals(1 + 2 + 3 + 4 + 5, summed.eval().scalar(), 1e-5);
+    }
+
+    @Test
+    public void doesSumSimpleAutoDiff() {
+        DoubleVertex a = new UniformVertex(new int[]{1, 5}, 0, 10);
+        a.setValue(new double[]{1, 2, 3, 4, 5});
+
+        DoubleVertex b = a.sum();
+
+        DoubleTensor dbdaForward = b.getDualNumber().getPartialDerivatives().withRespectTo(a);
+        DoubleTensor dbdaReverse = Differentiator.reverseModeAutoDiff(b, a).withRespectTo(a);
+
+        DoubleTensor expectedDbDa = DoubleTensor.create(new double[]{1, 1, 1, 1, 1}, 1, 1, 1, 5);
+
+        assertThat(dbdaForward, equalTo(expectedDbDa));
+        assertThat(dbdaReverse, equalTo(expectedDbDa));
+    }
+
+    @Test
+    public void canDoSumAutoDiffWhenSumIsNotWrtOrOf() {
+        DoubleVertex a = new UniformVertex(new int[]{2, 3}, 0, 10);
+        a.setValue(DoubleTensor.arange(0, 6).reshape(2, 3));
+
+        DoubleVertex d = a.sum();
+
+        DoubleVertex e = new UniformVertex(new int[]{2, 2}, 0, 10);
+        e.setValue(DoubleTensor.arange(4, 8).reshape(2, 2));
+
+        DoubleVertex f = d.times(e);
+
+        DoubleTensor dfdaForward = f.getDualNumber().getPartialDerivatives().withRespectTo(a);
+
+        PartialDerivatives dfdx = Differentiator.reverseModeAutoDiff(f, a, e);
```

Comment:
nit: can we just have `Differentiator.reverseModeAutoDiff(f, a);` instead - so it's clearer we're just after wrtA

Reply:
Yup. Good point.